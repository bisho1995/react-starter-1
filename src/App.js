import React from 'react'
import { hot } from 'react-hot-loader'

const count_red = {
  color: 'red'
}

const count_green = {
  color: 'blue'
}
class App extends React.Component {
  state = {
    count: 0
  }
  render() {
    return (
      <React.Fragment>
        <h1>
          {' '}
          Hello World there it is commit should be rejected ok, state should not
          be lost
        </h1>
        <button
          onClick={() => {
            this.setState(prevState => {
              return { count: prevState.count + 1 }
            })
          }}
        >
          {' '}
          Add{' '}
        </button>
        <button
          onClick={() => {
            this.setState(prevState => {
              return { count: prevState.count - 1 }
            })
          }}
        >
          {' '}
          Sub{' '}
        </button>
        <h2 style={this.state.count > 5 ? count_green : count_red}>
          Count = {this.state.count}
        </h2>
      </React.Fragment>
    )
  }
}

const hotFunction = hot(module)
export default hotFunction(App)
