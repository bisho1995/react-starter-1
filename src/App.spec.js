import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import { render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import App from './App'

describe('App', () => {
  it('Renders without error', () => {
    render(<App />)
  })
})

it('renders correctly', () => {
  const tree = renderer.create(<App />).toJSON()
  expect(tree).toMatchSnapshot()
})
