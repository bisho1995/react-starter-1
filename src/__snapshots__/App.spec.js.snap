// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <h1>
     
    Hello World there it is commit should be rejected ok, state should not be lost
  </h1>,
  <button
    onClick={[Function]}
  >
     
    Add
     
  </button>,
  <button
    onClick={[Function]}
  >
     
    Sub
     
  </button>,
  <h2
    style={
      Object {
        "color": "red",
      }
    }
  >
    Count = 
    0
  </h2>,
]
`;
